### PR TITLE
SOC-952 | Create reply method

### PR DESCRIPTION
**Pull request by kvas-damian**
Usage
```
this.get('model').createReply({
               threadId : "2614587795259262672",
               body: "test",
               creatorId: 5428365,
               siteId: 3035
            });
```

### CODE REVIEW

**Comment by bkoval** [comment]
LGTM :package:

File: front/scripts/main/models/DiscussionPostModel.ts [hunk 1]
```diff
@@ -51,12 +51,36 @@ App.DiscussionPostModel = Em.Object.extend(App.DiscussionErrorMixin, {
...
+				error: (err: any): void => {
 					this.setErrorProperty(err, this);
 					resolve(this);
 				}
 			});
 		});
+	},
+
+	createReply(replyData: any) {
+		this.setFailedState(false, this);
```
Comment (rafalkalinski) [inline front/scripts/main/models/DiscussionPostModel.ts]: Did you consider $.ajax beforeSend parameter for setting that?

File: front/templates/main/discussion/forum.hbs [hunk 2]
```diff
@@ -53,7 +53,7 @@
 					postsDisplayed=model.posts.length
 					totalPosts=model.totalPosts
 					siteHeadPinned=siteHeadPinned
-					shouldShowPostError=model.postFailed
+					shouldShowPostError=model.requestFailed
```
Comment (rafalkalinski) [inline front/templates/main/discussion/forum.hbs]: So DiscussionErrorMixin sets the property on the model, which is not defined in the model itself and we are passing it to the forum-wrapper component as any regular model property. Something is wrong here.

File: front/scripts/main/mixins/DiscussionErrorMixin.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ App.DiscussionErrorMixin = Em.Mixin.create({
 		Em.$('body').addClass(this.errorClass);
 	},
 
-	setPostFailedState(hasFailed: boolean, model: any): void {
-		model.set('postFailed', hasFailed);
+	setFailedState(hasFailed: boolean, model: any): void {
```
Comment (rafalkalinski) [inline front/scripts/main/mixins/DiscussionErrorMixin.ts]: Currently it has setErrorProperty and setFailedState methods, and it sets 'notFoundError' or 'connectionError' or 'requestFailed' properties on the model, when the request fails. 
If it's to much to handle it in this ticket scope, I'd be happy to file a separate ticket.

File: front/templates/main/discussion/forum.hbs (hunk 2 above)
Comment (bkoval) [inline front/templates/main/discussion/forum.hbs]: Please, be more clear about a specific change you would like to see here. Do you mean adding a variable definition to the model class?

File: front/scripts/main/mixins/DiscussionErrorMixin.ts (hunk 3 above)
Comment (bkoval) [inline front/scripts/main/mixins/DiscussionErrorMixin.ts]: Please, file a ticket to refine the whole error handling in discussions :)

File: front/scripts/main/models/DiscussionPostModel.ts (hunk 1 above)
Comment (kvas-damian) [inline front/scripts/main/models/DiscussionPostModel.ts]: From jQuery documentation:
> A pre-request callback function that can be used to modify the jqXHR (in jQuery 1.4.x, XMLHTTPRequest) object before it is sent. Use this to set custom headers, etc.  [...]

Are we sure we still want to use it?

Comment (rafalkalinski) [inline front/scripts/main/models/DiscussionPostModel.ts]: I would :)

File: front/templates/main/discussion/forum.hbs (hunk 2 above)
Comment (rafalkalinski) [inline front/templates/main/discussion/forum.hbs]: For start that would be good.

File: front/scripts/main/models/DiscussionPostModel.ts (hunk 1 above)
Comment (kvas-damian) [inline front/scripts/main/models/DiscussionPostModel.ts]: What does it change?

File: front/scripts/main/mixins/DiscussionErrorMixin.ts (hunk 3 above)
Comment (rafalkalinski) [inline front/scripts/main/mixins/DiscussionErrorMixin.ts]: SOC-1648

**Comment by rafalkalinski** [comment]
LGTM

File: front/scripts/main/models/DiscussionPostModel.ts (hunk 1 above)
Comment (rafalkalinski) [inline front/scripts/main/models/DiscussionPostModel.ts]: Nothing. But from my perspective beforeSend was mentioned to do things before the request is fired.

**Pull request merged**